scipy/optimize/_hessian_update_strategy.py

```diff
@@ -0,0 +1,338 @@
+"""quasi-Newton update strategies."""
+
+from __future__ import division, print_function, absolute_import
+import numpy as np
+from numpy.linalg import norm
+from scipy.linalg import get_blas_funcs
+
+class HessianUpdateStrategy(object):
+    """Virtual interface for implementing Hessian update strategies.
+
+    Should implement four methods: ``initialize``, ``update``,
+    ``dot`` and ``get_matrix``.
+
+    Notes
+    -----
+    Any instance of a class that implements this interface,
+    can be accepted by the method ``minimize`` and used by
+    the compatible solvers to approximate the Hessian (or
+    inverse Hessian) used by the optimization algorithms.
+    """
+
+    def initialize(self, n, approx_type):
+        """Initialize internal matrix.
```

Comment:
I think using "the internal matrix" everywhere is a good idea.

Reply:
I like it. I will try to include this nomenclature everywhere.